spec/controllers/results_controller_spec.rb

```diff
@@ -1201,12 +1213,21 @@ def self.test_unauthorized(route_name)
     before(:each) { sign_in ta }
     [:set_released_to_students].each { |route_name| test_unauthorized(route_name) }
     context 'accessing edit' do
-      before :each do
-        get :edit, params: { course_id: course.id, id: incomplete_result.id }, xhr: true
+      context 'when assigned to grade the given group\'s work' do
+        before :each do
+          get :edit, params: { course_id: course.id, id: incomplete_result.id }, xhr: true
+        end
+        test_no_flash
+        it { expect(response).to render_template('edit') }
+        it { expect(response).to have_http_status(:success) }
+      end
+      context 'when not assigned to grade the given group\'s work' do
+        let(:grouping) { create :grouping_with_inviter, assignment: assignment, inviter: student }
```

Comment:
Following from my previous comment, this `let` will no longer be necessary. Instead, you should add the `let!` to the "when assigned to grade" context.

(Similarly for other contexts below.)